posts/2021/jupyter-based-diagnostics-overview.ipynb

```diff
@@ -0,0 +1,434 @@
+    "\n",
+    "If you are interested in how we built the catalog visualization, I encourage you to check out the [\"Creating Model Documentation Using Jupyterbook and Intake-ESM\" blog post](https://ncar.github.io/esds/posts/2021/model_documentation_jupyterbook/)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using our Configuration File in the Notebooks\n",
+    "\n",
+    "At the top of every notebook, we call \n",
+    "\n",
+    "```python\n",
+    "from config import analysis_config\n",
+    "```\n",
+    "\n",
+    "which access the `analysis_config` variable from the `config.py` script:\n",
+    "\n",
+    "```python\n",
+    "import yaml\n",
+    "\n",
+    "with open(\"analysis_config.yml\", mode=\"r\") as fptr:\n",
+    "    analysis_config = yaml.safe_load(fptr)\n",
+    "\n",
+    "analysis_config['all_variables'] = analysis_config['variables']['physics'] + analysis_config['variables']['biogeochemistry']\n",
+    "analysis_config['all_cases'] = [analysis_config['reference_case_name']] + analysis_config['compare_case_name'] \n",
+    "```\n",
+    "\n",
+    "This allows us to access a dictionary with the configuration set in our `analysis_config.yml` file"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## How to Parameterize Notebooks\n",
+    "An important requirement for a diagnostic package is that it be \"parameterizable\" such that a user can input a list of variables, and the package is able to \"inject\" those variables into the analysis. Here, we use the `variables` section of the `analysis_config.yaml` file:\n",
+    "\n",
+    "```yaml\n",
+    "variables:\n",
+    "    physics:\n",
+    "        - TEMP\n",
+    "        - SALT\n",
+    "        - HMXL\n",
+    "        - BSF\n",
+    "        - IFRAC\n",
+    "        - SHF\n",
+    "        - SFWF\n",
+    "        - SHF_QSW\n",
+    "    biogeochemistry:\n",
+    "        - FG_CO2\n",
+    "        - PO4\n",
+    "        - photoC_TOT_zint\n",
+    "        - photoC_sp_zint\n",
+    "        - photoC_diat_zint\n",
+    "        - photoC_diaz_zint\n",
+    "        - POC_FLUX_100m\n",
+    "        - diaz_Nfix\n",
+    "        - SiO2_PROD\n",
+    "        - CaCO3_FLUX_100m\n",
+    "        - SiO2_FLUX_100m\n",
+    "        - ALK\n",
+    "        - DIC\n",
+    "        - O2\n",
+    "        - DOC\n",
+    "        - DOCr\n",
+    "        - NH4\n",
+    "        - NO3\n",
+    "        - SiO3\n",
+    "```\n",
+    "These variables can then be \"subsituted\" into notebooks using [papermill](https://papermill.readthedocs.io/en/latest/index.html) or [nbformat](https://nbformat.readthedocs.io/en/latest/).\n",
+    "\n",
+    "**Make sure you install papermill before adding these tags**\n",
+    "\n",
+    "```bash\n",
+    "python3 -m pip install papermill\n",
```

Comment:
Why is `papermill` not on conda-forge?

Reply:
Its available from both - I changed this line to 
```bash
conda install -c conda-forge papermill
```